File: CPE-1star/UVA10193-All You Need Is Love.py
#將兩字串轉十進位，找公因數，若等於1則fail 
import math
while True:
	try:
		N=int(input())
		no=1
		for i in range(N):
			s1=int(input())
			s2=int(input())
			bs=0
			bs1,bs2=0,0
			while s1>0:
				bs1+=(s1%10)*(2**bs)
				s1=s1//10
				bs+=1
			bs=0
			while s2>0:
				bs2+=(s2%10)*(2**bs)
				s2=s2//10
				bs+=1
			m=math.gcd(bs1,bs2)
			if m!=1:
				print("Pair #%d: All you need is love!"%(no))
			else:
				print("Pair #%d: Love is not all you need!"%(no))
			no+=1
	except EOFError:
		break

File: CPE-1star/UVA10193-All You Need Is Love.cpp
#include <iostream>
#include<cmath>
using namespace std;
int gcd(int a,int b){
    if(a==0 & b!=0)
        return a;
    if(b==0)
        return a;
    else
        return gcd(b,a%b);
}
int main()
{
    int n,no=1;
    while(cin>>n){
        for(int i=0;i<n;i++){
            long long a,b,m;
            cin>>a;
            cin>>b;
            long long bs=0,bs1=0,bs2=0;
            while(a>0){
                bs1+=(a%10)*(int)pow(2,bs);
                a=a/10;
                bs+=1;
            }
            bs=0;
            while(b>0){
                bs2+=(b%10)*(int)pow(2,bs);
                b=b/10;
                bs+=1;
            }
            m=gcd(bs1,bs2);
            if(m!=1)
                cout<<"Pair #"<<no<<": All you need is love!"<<'\n';
            else    
                cout<<"Pair #"<<no<<": Love is not all you need!"<<'\n';
            no+=1;
        }
    }
    return 0;
}
